linkcheck/configuration/__init__.py

```diff
@@ -543,54 +535,39 @@ def get_kde_ftp_proxy():
 
 def get_kde_config_dir():
     """Return KDE configuration directory or None if not found."""
-    kde_home = get_kde_home_dir()
-    if not kde_home:
-        # could not determine the KDE home directory
-        return
-    return kde_home_to_config(kde_home)
-
-
-def kde_home_to_config(kde_home):
-    """Add subdirectories for config path to KDE home directory."""
-    return os.path.join(kde_home, "share", "config")
-
-
-def get_kde_home_dir():
-    """Return KDE home directory or None if not found."""
     if os.environ.get("KDEHOME"):
-        kde_home = os.path.abspath(os.environ["KDEHOME"])
+        home = os.environ.get("KDEHOME")
     else:
         home = os.environ.get("HOME")
-        if not home:
-            # $HOME is not set
-            return
-        kde3_home = os.path.join(home, ".kde")
-        kde4_home = os.path.join(home, ".kde4")
-        if shutil.which("kde4-config"):
-            # kde4
-            kde3_file = kde_home_to_config(kde3_home)
-            kde4_file = kde_home_to_config(kde4_home)
-            if os.path.exists(kde4_file) and os.path.exists(kde3_file):
-                if fileutil.get_mtime(kde4_file) >= fileutil.get_mtime(kde3_file):
-                    kde_home = kde4_home
-                else:
-                    kde_home = kde3_home
-            else:
-                kde_home = kde4_home
-        else:
-            # kde3
-            kde_home = kde3_home
-    return kde_home if os.path.exists(kde_home) else None
+    if not home:
+        log.debug(LOG_CHECK, "KDEHOME and HOME not set")
+        return
+    if shutil.which("kde4-config"):
+        kde_config_dir = os.path.join(home, ".kde4", "share", "config")
+    else:
+        # KDE 5
+        kde_config_dir = os.path.join(home, ".config")
```

Comment:
What if the user is running KDE 5, but they still happen to have a kde4-config binary installed from some outdated package?  It would make more sense to look for the actual config file in both directories, IMHO, and prefer the newer one if it exists.

Reply:
OK. Dropped the check for kde4-config and swapped the tests around to try KDE 5 first (most likely in 2020).